src/io/fastq.rs

```diff
@@ -3,24 +3,103 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! FastQ reading.
+//! Structs and trait to read and write file in FASTQ format.
 //!
 //! # Example
 //!
+//! ## Read
+//!
+//! In this example, we parse a fastq file from stdin and compute some statistics
+//!
 //! ```
+//! use std::io;
 //! use bio::io::fastq;
-//! use bio::io::fastq::FastqRead;
+//! let mut reader = fastq::Reader::new(io::stdin());
+//!
+//! let mut nb_reads = 0;
+//! let mut nb_bases = 0;
+//!
+//! for result in reader.records() {
+//!     let record = result.expect("Error during fastq record parsing");
+//!
+//!     nb_reads += 1;
+//!     nb_bases += record.seq().len();
+//! }
+//!
+//! println!("Number of reads: {}", nb_reads);
+//! println!("Number of bases: {}", nb_bases);
+//! ```
+//!
+//! We can also use a `while` loop to iterate over records
+//! ```
 //! use std::io;
+//! use bio::io::fastq;
+//! let mut records = fastq::Reader::new(io::stdin()).records();
+//!
+//! let mut nb_reads = 0;
+//! let mut nb_bases = 0;
+//!
+//! while let Some(Ok(record)) = records.next() {
+//!     nb_reads += 1;
+//!     nb_bases += record.seq().len();
+//! }
+//!
+//! println!("Number of reads: {}", nb_reads);
+//! println!("Number of bases: {}", nb_bases);
+//! ```
+//!
+//! ## Write
+//!
+//! In this example we generate 10 random sequences with length 100 and write them to sdtout.
```

Comment:
I think you may have to apply your sed magic again ;D
```suggestion
//! In this example we generate 10 random sequences with length 100 and write them to stdout.
```